tests/performance/utils.go

```diff
@@ -0,0 +1,145 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2021 Red Hat, Inc.
+ *
+ */
+
+package performance
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net"
+	"os"
+	"path"
+	"strconv"
+	"time"
+
+	e2emetrics "kubevirt.io/kubevirt/tests/performance/metrics"
+	"kubevirt.io/kubevirt/tests/performance/perftype"
+)
+
+// LogPerfData prints the perf data
+func LogPerfData(p *perftype.PerfData, perfType string) {
+	// Notice that we must make sure the perftype.PerfResultEnd is in a new line.
+	if str := prettyPrintJSON(p); str != "" {
+		fmt.Printf("\n%s %s%s\n", perftype.PerfResultTag, str, perftype.PerfResultEnd)
+	}
+	if CreatePerfReport {
+		dumpDataToFile(p, p.Labels, perfType)
+	}
+}
+
+// prettyPrintJSON converts metrics to JSON format.
+func prettyPrintJSON(metrics interface{}) string {
+	output := &bytes.Buffer{}
+	if err := json.NewEncoder(output).Encode(metrics); err != nil {
+		fmt.Printf("Error building encoder: %v", err)
+		return ""
+	}
+	formatted := &bytes.Buffer{}
+	if err := json.Indent(formatted, output.Bytes(), "", "  "); err != nil {
+		fmt.Printf("Error indenting: %v", err)
+		return ""
+	}
+	return formatted.String()
+}
+
+// DumpDataToFile inserts the current timestamp into the labels and writes the data for the test into the file with the specified sufix.
+func dumpDataToFile(data interface{}, labels map[string]string, sufix string) {
```

Comment:
Performance data output seems like a good abstraction.  In other words, if we capture any data for perf or scale, the output can end up in prometheus, stdout, to a file, ect..., depending on how a user configures kubevirt.  Can we scope out an interface that would work for all use cases?

Reply:
I tried to create the `PerfData` and `DataItem` structures very generic, so I believe it can support different cases.

```
// DataItem is the data point.
type DataItem struct {
	Data map[string]float64 `json:"data"`
	Unit string `json:"unit"`
	Labels map[string]string `json:"labels,omitempty"`
}

// PerfData contains all data items generated in current test.
type PerfData struct {
	Version   string     `json:"version"`
	DataItems []DataItem `json:"dataItems"`
	Labels map[string]string `json:"labels,omitempty"`
}
```

The hard constraint is that the values must always be `float64`, but the unit and names (i.e., labels) are any string value.

Reply:
Btw, [here](https://docs.google.com/document/d/1g5tjv1GcxNsw0pDmToz6z5NRj9dVZTnbsAOvjj8Qdek/edit) is an example of the output of this test.